src/test/java/Homework21.java

```diff
@@ -0,0 +1,66 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import page.HomePage;
+import page.LoginPage;
+import page.PlaylistPage;
+import pageFactory.AllSongsPage;
+
+import static org.testng.Assert.assertEquals;
+
+public class Homework21 extends BaseTest {
+
+    String UpdatedName = "Good Morning";
+
+    @Test
+    public void newPlaylistName() {
+
+        // GIVEN; User is on login with availed email and password home page is displayed
+        HomePage homePage = new HomePage(driver);
+        LoginPage loginPage = new LoginPage(driver);
+        AllSongsPage allSongsPage =new AllSongsPage(driver);
+        PlaylistPage playlistPage = new PlaylistPage(driver);
+        loginPage.login();
+
+        //Given: User is able to change playlist name
+        String UpdatedName = "Good Morning";
+        doubleClickPlaylist();
+
+        //THEN: Updated message is displayed
+        playlistPage.renamePlaylist(UpdatedName);
+
+    }
+
+
+    public void renamePlaylist() {
+
+        WebElement playlistFiled = wait.until(ExpectedConditions.presenceOfElementLocated
+                (By.cssSelector("[name='name']")));
+        playlistFiled.sendKeys(Keys.chord(Keys.CONTROL, "A",Keys.BACK_SPACE));
+        playlistFiled.sendKeys(UpdatedName);
+        playlistFiled.sendKeys(Keys.ENTER);
+
+
+
+    }
+
+
+    public String successMessage() {
+        WebElement notification = wait.until(ExpectedConditions.visibilityOfElementLocated
+                (By.cssSelector("div.success.show")));
+        return notification.getText();
+
+
+    }
+
+    public void doubleClickPlaylist() {
+
+        WebElement playlistLazyDays = wait.until(ExpectedConditions.visibilityOfElementLocated
+                (By.cssSelector(".playlist:nth-child(4)")));
+        actions.doubleClick(playlistLazyDays).perform();
+
+    }
```

Comment:
In the newPlaylistName test method, there are helper methods called renamePlaylist(), successMessage(), and doubleClickPlaylist(). **These helper methods should be placed in the appropriate page classes according to the Page Object Model (POM) principles.** 

1) The renamePlaylist(String updatedName) method should be in the PlaylistPage class. It handles renaming the playlist.

2) The successMessage() method should be in the PlaylistPage class. It retrieves and returns the success message after renaming the playlist.

3) The doubleClickPlaylist() method should be in the PlaylistPage class. It handles double-clicking on the playlist to select it.

![Screenshot_3](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/7aa05527-c06a-4422-addc-cca0d1cb5081)

**Please check the test structure in the class LoginTest again carefully.**

**One important rule: Helper methods should be placed in the appropriate page classes according to the Page Object Model (POM) principles.**  

For example 
https://www.selenium.dev/documentation/test_practices/encouraged/page_object_models/